### PR TITLE
refactor(utils): migrate remaining utils to TypeScript

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

Some utils were written in JS still, so now they can be written in TypeScript instead

**Result**

All utils, except renderTemplate are in TS now
removed unused utils folder

### CODE REVIEW

File: src/types/templates.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+type StringTemplate = string;
+
+type FunctionTemplate = (data: unknown) => string;
+
+export type Templates = {
+  [key: string]: StringTemplate | FunctionTemplate;
```
Comment (francoischalifour) [inline src/types/templates.ts]: Templates are already typed in a generic way:

https://github.com/algolia/instantsearch.js/blob/c7c53de037959070422c7be85882423078dc1253/src/types/widget.ts#L194-L196

Comment (Haroenv) [reply src/types/templates.ts]: ah I missed it. I'll use that one then

File: src/lib/utils/prepareTemplateProps.ts [hunk 2]
```diff
@@ -1,6 +1,28 @@
 import uniq from './uniq';
+import { Template } from '../../types';
 
-function prepareTemplates(defaultTemplates = {}, templates = {}) {
+type TemplatesConfig = object;
+
+type Templates = {
+  [key: string]: Template;
```
Comment (francoischalifour) [inline src/lib/utils/prepareTemplateProps.ts]: We don't leverage the generics here but not sure we need them.

Comment (Haroenv) [reply src/lib/utils/prepareTemplateProps.ts]: Yes, as far as I can tell we don't know anything about the type in this place, so I left it like this. Future usage (unlikely) can precise this further

**Pull request merged**